core-library/src/main/java/org/silverpeas/core/admin/service/Admin.java

```diff
@@ -4799,22 +4817,26 @@ private String checkOutGroups(String domainId, GroupDetail[] existingGroups, Gro
     SearchCriteriaDAOFactory factory = SearchCriteriaDAOFactory.getFactory();
     GroupSearchCriteriaForDAO criteria = factory.getGroupSearchCriteriaDAO();
     if (searchCriteria.isCriterionOnComponentInstanceIdSet()) {
-      List<String> listOfRoleNames = new ArrayList<>();
+      List<String> listOfRoleNames = Collections.emptyList();
       if (searchCriteria.isCriterionOnRoleNamesSet()) {
         listOfRoleNames = Arrays.asList(searchCriteria.getCriterionOnRoleNames());
       }
-      ComponentInst instance = getComponentInst(searchCriteria.getCriterionOnComponentInstanceId());
+      SilverpeasComponentInstance instance =
+          getComponentInstance(searchCriteria.getCriterionOnComponentInstanceId());
       if (!listOfRoleNames.isEmpty() || !instance.isPublic()) {
-        List<ProfileInst> profiles;
-        if (searchCriteria.isCriterionOnResourceIdSet()) {
-          profiles = getProfileInstsFor(searchCriteria.getCriterionOnResourceId(), instance.getId());
-        } else {
-          profiles = instance.getAllProfilesInst();
-        }
         List<String> roleIds = new ArrayList<>();
-        for (ProfileInst aProfile : profiles) {
-          if (listOfRoleNames.isEmpty() || listOfRoleNames.contains(aProfile.getName())) {
-            roleIds.add(aProfile.getId());
+        if (!instance.isPersonal()) {
+          List<ProfileInst> profiles;
+          if (searchCriteria.isCriterionOnResourceIdSet()) {
+            profiles =
+                getProfileInstsFor(searchCriteria.getCriterionOnResourceId(), instance.getId());
+          } else {
+            profiles = getComponentInst(instance.getId()).getAllProfilesInst();
+          }
+          for (ProfileInst aProfile : profiles) {
+            if (listOfRoleNames.isEmpty() || listOfRoleNames.contains(aProfile.getName())) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)